lib/mixlib/install/backend/base.rb

```diff
@@ -94,18 +94,19 @@ def filter_artifacts(artifacts)
           artifacts.each do |a|
             return a if a.platform_version == options.platform_version
 
-            # We skip the artifacts produced for windows since their platform
-            # version is always set to 2008r2 which breaks our `to_f` comparison.
-            next if a.platform == "windows"
-
             # Calculate the closest compatible version.
             # For an artifact to be compatible it needs to be smaller than the
             # platform_version specified in options.
             # To find the closest compatible one we keep a max of the compatible
             # artifacts.
+
+            # Remove "r2" from artifacts produced for windows since platform
+            # versions with that ending break `to_f` comparisons.
+            current_platform_version = a.platform_version.chomp("r2").to_f
```

Comment:
Would it be better to just do a gsub and remove all non-numeric characters? Something like this should work:

```ruby
=> true
irb(main):003:0> "2008r2".gsub(/[^\d,\.]/, '')
=> "20082"
irb(main):004:0> "2008".gsub(/[^\d,\.]/, '')
=> "2008"
```